src/App/Components/Common/NavProfile/Profile.js

```diff
@@ -0,0 +1,126 @@
+/* eslint-disable jsx-a11y/click-events-have-key-events */
+/* eslint-disable jsx-a11y/no-noninteractive-element-interactions */
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import _ from 'lodash';
+import PropTypes from 'prop-types';
+import actions from '../../../../Redux/Actions/notifications';
+import defaultData from '../../../../configs/urls';
+import Triangle from '../Triangle';
+
+const { show, hide } = actions;
+
+export class Profile extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      isShown: false,
+      profile: {},
+      notifications: [],
+      notificationsBubble: ''
+    };
+  }
+
+  componentWillReceiveProps({
+    isShown, profile, configs, data
+  }) {
+    this.setState({ isShown });
+    if (!_.isEmpty({ ...profile, ...configs })) this.setState({ profile });
+    else this.setState({ profile: {} });
+    if (data && !_.isEmpty(data)) {
```

Comment:
if (data && !_.isEmpty(data))
it can be done only by `!_.isEmpty(data)`

Reply:
`!_.isEmpty(data)` was raising an error when receiving undefined...  But also as I'm using data in the render method, I had to use it in my state only when it is not undefined